Sequenced pool synonym for legacy pool

matthew1001 opened the pull request:
## PR description
Introduces the new `sequenced` synonym for the `legacy` transaction pool, as discussed in https://github.com/hyperledger/besu/issues/6211.

The PR adds enough for users to start switching to the `--tx-pool=sequenced` option ahead of removal of the `legacy` pool in the future.

I've added one or two additional unit tests to check that `legacy` options still work with the `sequenced` pool, but haven't done anything more comprehensive than that because the future PR to ~~deprecate~~ remove the `legacy` pool will refactor all of the existing tests to use `sequenced` anyway.

## Fixed Issue(s)
Fixes https://github.com/hyperledger/besu/issues/6211

matthew1001 commented:
Re. docs, I think the https://besu.hyperledger.org/public-networks/concepts/transactions/pool topic probably ought to have a new sub-heading for the sequenced pool. We didn't have one before for the legacy pool but with recent changes to the legacy pool we should try to describe the use-cases the sequenced pool is aimed at and how it differs from the layered pool.

I'd be happy to provide some words to help with that docs topic.

macfarla commented on besu/src/main/java/org/hyperledger/besu/cli/options/TransactionPoolOptions.java:
```diff
@@ -295,14 +296,14 @@ public void validate(
         commandLine,
-        "Could not use layered transaction pool options with legacy implementation",
-        !txPoolImplementation.equals(LEGACY),
+        "Could not use layered transaction pool options with legacy or sequenced implementation",
```
```suggestion
        "Could not use layered transaction pool options with legacy (sequenced) implementation",
```

matthew1001 replied:
Yeah it's difficult balancing the fact that we don't want to make the sequenced pool sound like it's the "legacy" option, even though in reality they are the same implementation. At some point the legacy option will be removed entirely so we'll update these messages to just talk about sequenced in the future. Personally I'd prefer to leave it as "legacy or sequenced" for now, but happy to discuss further and we can always raise another PR to re-word if we need to?

macfarla reviewed (approved):
LGTM . non-blocking suggestion re warning message text - just to make it clearer that they are not separate implementations

Also prob need a changelog entry

matthew1001 commented:
> LGTM . non-blocking suggestion re warning message text - just to make it clearer that they are not separate implementations
> 
> Also prob need a changelog entry

Thanks for the comments @macfarla. I've added a reply about the warning message wording. Re. change log entry, I've added a one-liner to it already. Did you want anything more substantial, like Fabio did for the layered pool in `23.10.0`?
